### PR TITLE
docs: switch to docus, rework all docs, add more docs

### LINKED ISSUE #53: Setup deployment of the docs page
**Goal**
In order to improve our DX, we want to publish are own extensive docs page. The docs have already been created in https://github.com/sidebase/docs and need to be transferred into the respective module repositories. 
Afterwards we hope to host each documentation with GitHub Pages, allowing them to be redeployed after every merge to the main branch. 

In order to do this we first need to check how we can correctly assign subdomains to a specific github page. In this case we want to use the subdomain https://nuxt-auth.sidebase.io to publish our docs. 

As far as I see, we just need to add a CNAME record for all subdomains of sidebase.io to point to the GitHub servers. Once done, we can verify the subdomain through the GitHub verification process to provide proof of ownership. 

**Links**
https://docs.github.com/en/pages/configuring-a-custom-domain-for-your-github-pages-site/about-custom-domains-and-github-pages
https://sean-coughlin.medium.com/how-to-host-a-site-with-a-subdomain-on-github-pages-33202341f711

**Tasks**

- [ ] Finish documentation (@zoey-kaiser for finishing, @BracketJohn for review)
- [ ] Setup CNAME record for the domain (@IronicUsername)
- [ ] Add documentation folder to `nuxt-auth` repo (@zoey-kaiser)
- [ ] Setup Github Actions to rebuild docs so we do not need to do it manually (optional, depending on how much extra work it presents, @IronicUsername)

### LINKED ISSUE #91: Add an `ignorePaths` option to the global middleware
### Describe the feature

Add an `ignorePaths` option to the global middleware, so that pages can be excluded from the global protetction without adding:
```ts
definePageMeta({ auth: false }) 
```
to them.

This approach has some advantages:
- it works if a project hast just a `app.vue`. Currently, in this case, the `app.vue` cannot be excluded as the `definePageMeta` macro does not work on it
- it may offer a more intuitive way for some users to configure the excluded paths directly in the `nuxt.config.ts`

### Additional information

Originally proposed here: https://github.com/sidebase/nuxt-auth/issues/81#issuecomment-1339004288

Interface proposal for config:
```ts
enableGlobalAppMiddleware: boolean | {
  enabled: boolean
  ignoresPaths: string[]
}
```

### PR DESCRIPTION
**Pull request by BracketJohn**
Contributes to #30
Contributes to #61
Closes #91
Closes #53

Checklist:
- [x] issue number linked above after pound (`#`)
    - replace "Closes " with "Contributes to" or other if this PR does not close the issue
- [x] manually checked my feature / checking not applicable
- [x] wrote tests / testing not applicable
- [x] attached screenshots / screenshot not applicable

### CODE REVIEW

File: docs/content/1.getting-started/4.getting-help.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Getting Help
+
+At some point, you may find that there's an issue you need some help with.
+
+But don't worry! We're a friendly community of developers and we'd love to help. Concretely this means to:
+- Checkout our docs here,
```
Comment (zoey-kaiser) [inline docs/content/1.getting-started/4.getting-help.md:6]: Do we want to add a link here?

File: docs/content/5.recipes/1.strapi.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+- [`auth`-key in `nuxt.config.ts`](/nuxt-auth/configuration/nuxt-config): Configure the module itself, e.g., where the auth-endpoints are, what origin the app is deployed to, ...
+- [NuxtAuthHandler](/nuxt-auth/configuration/nuxt-auth-handler): Configure the authentication behavior, e.g., what authentication providers to use
+
+For a production deployment, you will have to at least set the:
+- `STRAPI_BASE_URL` Strapi base URL for all API endpoints by default http://localhost:1337
+
+1. Create a `.env` file with the following lines:
+```env
+// Strapi v4 url, out of the box
+ ORIGIN=http://localhost:3000
```
Comment (zoey-kaiser) [inline docs/content/5.recipes/1.strapi.md:17]: There is one extra space before all of these env variables.

File: docs/content/5.recipes/2.directus.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        ...token,
+      };
+
+      return session;
+    },
+  },
+});
+```
+
+This was contributes by @madsh93 from Github here:
```
Comment (zoey-kaiser) [inline docs/content/5.recipes/2.directus.md:199]: Do we want to link the username to the GitHub Profile?

File: docs/content/5.recipes/3.custom-session-data.md [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+declare module "next-auth" {
+  interface Session {
+    // New access token field we added to the session
+    accessToken: string
+    user: DefaultSession["user"]
+  }
+}
+```
+
+Thanks to @JoaoPedroAS51 from Github for providing this information here: https://github.com/sidebase/nuxt-auth/issues/61#issuecomment-1330747199
```
Comment (zoey-kaiser) [inline docs/content/5.recipes/3.custom-session-data.md:57]: Do we want to link the username to the profile?

**Review by zoey-kaiser — CHANGES_REQUESTED** [review]

File: docs/content/1.getting-started/4.getting-help.md (hunk 1 above)
Comment (BracketJohn) [reply docs/content/1.getting-started/4.getting-help.md:6]: I've clarified a bit, but this is already talking about uitself :D

Comment (zoey-kaiser) [reply docs/content/1.getting-started/4.getting-help.md]: True!

**Pull request merged**